### ISSUE TITLE
Missing LDAPS support

### ISSUE THREAD

**Issue by ethanmye-rs** [canonical/candid#168]
When trying to set the LDAP identity provider, the LDAPS protocol is not supported and fails to parse.

$ sudo snap logs candid
2023-02-23T00:01:51Z systemd[1]: Started Service for snap application candid.candidsrv.
2023-02-23T00:01:51Z candid.candidsrv[809084]: STOP cannot read configuration: cannot parse "/var/snap/candid/799/config.yaml": cannot unmarshal ldap configuration: unsupported scheme "ldaps"
2023-02-23T00:01:51Z systemd[1]: snap.candid.candidsrv.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
2023-02-23T00:01:51Z systemd[1]: snap.candid.candidsrv.service: Failed with result 'exit-code'.
2023-02-23T00:01:51Z systemd[1]: snap.candid.candidsrv.service: Scheduled restart job, restart counter is at 5.
2023-02-23T00:01:51Z systemd[1]: Stopped Service for snap application candid.candidsrv.
2023-02-23T00:01:51Z systemd[1]: snap.candid.candidsrv.service: Start request repeated too quickly.
2023-02-23T00:01:51Z systemd[1]: snap.candid.candidsrv.service: Failed with result 'exit-code'.
2023-02-23T00:01:51Z systemd[1]: Failed to start Service for snap application candid.candidsrv.

This results in the app quickly hitting the restart counter and needing to manually start and stop the application to get candid working again after editing the config.

To reproduce, copy the default LDAP config from [here](https://github.com/canonical/candid/blob/master/docs/configuration.md) and replace ldap:// with ldaps://

**Comment by patrikarlos** [comment]
Did you find a workaround?

**Comment by zhhuabj** [comment]
It seems OpenLDAP's official guidance [1] recommends using StartTLS rather than ldaps:// for this use case.

"ldaps:// is deprecated in favor of Start TLS [RFC2830]"

Since Candid already supports StartTLS, the current implementation already meets the standard for TLS/SSL encryption. Furthermore, once initiated, there is no difference between ldaps:// and StartTLS, they share the same configuration options and result in like security services being established [3].

Given that StartTLS is the industry-standard approach and is already supported, should we propose closing this case?

[1] https://www.openldap.org/faq/data/cache/605.html
[2] https://www.openldap.org/faq/data/cache/185.html

**Comment by patrikarlos** [comment]
Does the current LDAP provider config support "StartTLS" option/argument?
https://github.com/canonical/candid/blob/master/docs/configuration.md#ldap

**Comment by zhhuabj** [comment]
> Does the current LDAP provider config support "StartTLS" option/argument? https://github.com/canonical/candid/blob/master/docs/configuration.md#ldap

Yes, candid supports starttls. Use the ldap:// prefix with port 389 (  For ldaps, use the ldaps:// prefix with port 636 ) and provide tls-cert and tls-key for candid server and ca-cert for candid client to enable tls.

**Comment by patrikarlos** [comment]
Please update the doc to match this info. Then close this?
